enzyme/test/Fortran/ReverseMode/square.f90

```diff
@@ -0,0 +1,38 @@
+! RUN: ifx -flto -O0 -c %s | %opt - %loadEnzyme -enzyme
+! RUN: ifx -flto -O1 -c %s | %opt - %loadEnzyme -enzyme
+! RUN: ifx -flto -O2 -c %s | %opt - %loadEnzyme -enzyme
+! RUN: ifx -flto -O3 -c %s | %opt - %loadEnzyme -enzyme
+
```

Comment:
Can you add a check line? Eg maybe through lli or run an executable?

Reply:
Yep I'll do that

Reply:
@wsmoses I think there is also a bug in what I'm doing above. So to fix that and add more checking I propose,
```
! RUN: ifx -flto -O0 -c  %s -o %t && %opt %t %loadEnzyme -enzyme -o %t1 && ifx -flto -O0 %t1 -o %t2 && %t2 | FileCheck %s
```
...
```
! CHECK: 9
! CHECK-NEXT: 6
``` 
Its more file heavy, but I can't find a way to have ifx compile from stdin. Also I went for running the exe since I'm expecting lli not to understand the intrinsics/runtime library functions later down the line when tests including those arise.

Reply:
Good question. Yes, ifx can print to stdout either by using `-o /dev/stdout` or I can use process substitution `ifx ... -o >(opt ...)`. I need to check if the second is valid in LIT. So I could replace the first temporary file.

Reply:
Also just to add to that, it doesn't support `-o - ` like in clang.